streams/src/test/java/org/apache/kafka/streams/integration/StandbyTaskFailOverIntegrationTest.java

```diff
@@ -0,0 +1,283 @@
+                    long timestamp = context.timestamp();
+                    final WindowStoreIterator<String> storeIterator = dedupStore.fetch(key, timestamp - WINDOW_SIZE.toMillis(), timestamp);
+                    if (storeIterator.hasNext()) {
+                        throw new IllegalStateException("Caught a duplicate key " + key);
+                    }
+                    dedupStore.put(key, value, timestamp);
+
+                    return new KeyValue<>(key, value);
+                }
+
+                @Override
+                public void close() {
+                }
+            }, STORE_NAME
+        ).peek((key, value) -> {
+            if (key.equals(poisonKey)) {
+                try {
+                    if (waitPoisonRecordReplication != null) {
+                        waitPoisonRecordReplication.await(15_000L, TimeUnit.MILLISECONDS);
+                    }
+                } catch (InterruptedException ignored) {
+                }
+                throw new IllegalStateException("Throw on key_two to trigger rebalance");
+            }
+        });
+
+        return new KafkaStreams(builder.build(),
+            props(numThreads, String.format("/tmp/kafka-streams/instance-%d/", instanceId)));
+    }
+
+    @After
+    public void shutdown() {
+        if (streamInstanceOne != null) {
+            streamInstanceOne.close(Duration.ofSeconds(30L));
+            streamInstanceOne.cleanUp();
+        }
+
+        if (streamInstanceTwo != null) {
+            streamInstanceTwo.close(Duration.ofSeconds(30L));
+            streamInstanceTwo.cleanUp();
+        }
+    }
+
+    private Properties props(final int numThreads, final String stateDirPath) {
+        final Properties streamsConfiguration = new Properties();
+        streamsConfiguration.put(StreamsConfig.APPLICATION_ID_CONFIG, appId);
+        streamsConfiguration.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
+        streamsConfiguration.put(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG, 0);
+        streamsConfiguration.put(StreamsConfig.NUM_STANDBY_REPLICAS_CONFIG, 1);
+        streamsConfiguration.put(StreamsConfig.NUM_STREAM_THREADS_CONFIG, numThreads);
+        streamsConfiguration.put(StreamsConfig.STATE_DIR_CONFIG, stateDirPath);
+        streamsConfiguration.put(StreamsConfig.PROCESSING_GUARANTEE_CONFIG, eosConfig);
+        streamsConfiguration.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.Integer().getClass());
+        streamsConfiguration.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.Integer().getClass());
+        // Set commit interval long to avoid actually committing the record.
+        streamsConfiguration.put(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG, Integer.MAX_VALUE);
+        streamsConfiguration.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        streamsConfiguration.put(ProducerConfig.LINGER_MS_CONFIG, 0);
+
```

Comment:
I've noticed there's only a single rebalance, and we need at least 3. maybe set 
`        streamsConfiguration.put(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, 1000);
        streamsConfiguration.put(ConsumerConfig.HEARTBEAT_INTERVAL_MS_CONFIG, 500);`
or similar